src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,88 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int hashCode = key == null ? 0 : key.hashCode();
+        int index = (hashCode & Integer.MAX_VALUE) % table.length;
```

Comment:
create separate methods 
```suggestion
        int bucketIndex = getBucketIndex(key);
```

```
private int hash(K key) {
        return Math.abs(Objects.hashCode(key));
    }

    private int getBucketIndex(K key) {
        return hash(key) % table.length;
    }
```